packages/workflow/src/Interfaces.ts

```diff
@@ -564,13 +564,23 @@ namespace ExecuteFunctions {
 			| 'resolveData';
 	}
 
+	namespace RecordReturning {
+		export type NodeParameter = 'additionalFields' | 'filters' | 'options' | 'updateFields';
+	}
+
 	export type GetNodeParameterFn = {
 		// @TECH_DEBT: Refactor to remove this barely used overload - N8N-5632
 		getNodeParameter<T extends { resource: string }>(
 			parameterName: 'resource',
 			itemIndex?: number,
 		): T['resource'];
 
+		getNodeParameter(
+			parameterName: RecordReturning.NodeParameter,
+			itemIndex: number,
+			fallbackValue?: IDataObject,
+			options?: IGetNodeParameterOptions,
+		): IDataObject;
```

Comment:
Ideally we should narrow down `IDataObject` in future, but too widely used to refactor in this PR.

Reply:
I'm certain any PR wanting to replace `IDataObject` with stronger types is going to me much larger than this PR 😮‍💨 🤦🏽